### ISSUE TITLE
Possibility of adapting to the go-emotion original multilabel task

### ISSUE THREAD

**Issue by zhangziyi1996** [dinobby/HypEmo#4]
Hello sir, just read your paper and I’m really fascinated by your good F1 numbers. However I want to apply your proposed model on the go-emotion original multi label setup. I made the following attempts and modifications:
1. Change the dataset to be the original go-emotion dataset
2. When calculating CE_loss, Replace Cross-entropy with Binary cross entropy
3. If a training instance has multiple labels, average every individual label’s hyperbolic arrays to get its label_vec. (I’m sorry if this step seems too straightforward and naive. I have little knowledge about the Hyperbolic space and I think it’s infeasible to create a hyperbolic array for every unique label combination as the variations are too many.)

Weirdly, I ended up with getting some very strange results. Specifically, the Poincaré loss term inside the total loss calculation becomes negative values from the end of the first epoch and onwards. The negative values grows smaller and smaller, ultimately causes the total loss term become negative. I’ve made some simple investigations and found that it’s due to the dot_product term(in Poincaré model compute_metrics() function) value becoming too large. 

Can you please suggest which part I did wrong? Should I also change some hyper parameters in the hyperbolic part to resolve negative loss issue? Do you think your model theoretically support multi label task and can be easily adapted to? 

Sorry for the lengthy questions. Really appreciate your time and assistance!
